From a9c1d37cf5ec789c966503f3218dbec7e00224d3 Mon Sep 17 00:00:00 2001
From: Yong Zhao <yong.zhao@amd.com>
Date: Fri, 13 May 2016 21:39:48 -0400
Subject: [PATCH 5458/5855] drm/amdkfd: Unmapping static queues when a process
 terminates

In the original code, the static queues were also unmapped when a
process terminates. However, when we tried to fix a multi-process
dispatch failure in SWDEV-89586, we excluded the static queues in
order not to reset the queues when unmapping. Now we include back
the static queues as the new interface of execute_queues_cpsch()
allows us to specify the queue filter.

Change-Id: I09959e96fc6ee2990ff80c251910dcc7c0c631c8
Signed-off-by: Yong Zhao <yong.zhao@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
index 5289427..26e6044 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
@@ -1460,7 +1460,7 @@ static int process_termination_cpsch(struct device_queue_manager *dqm,
 		}
 	}
 
-	retval = execute_queues_cpsch(dqm, false);
+	retval = execute_queues_cpsch(dqm, true);
 
 	/* lastly, free mqd resources */
 	list_for_each_entry_safe(q, next, &qpd->queues_list, list) {
-- 
2.7.4

